### ISSUE TITLE
Bad file descriptor error with mochawesome reporter v7

### ISSUE THREAD

**Issue by cexbrayat** [cypress-io/cypress#18786]
### Current behavior

Mochawesome v7 just released, and using it with Cypress v8.7 and the following configuration:
```json
{
  "reporter": "mochawesome",
  "reporterOptions": {
    "overwrite": false
  }
}
```

produces the following error:

```
[28541:1104/130233.876580:FATAL:scoped_file.cc(43)] Check failed: . : Bad file descriptor (9)
The Test Runner unexpectedly exited via a exit event with signal SIGTRAP

Please search Cypress documentation for possible solutions:

https://on.cypress.io

Check if there is a GitHub issue describing this crash:

https://github.com/cypress-io/cypress/issues

Consider opening a new issue.

----------

Platform: darwin (20.6.0)
Cypress Version: 8.7.0
```

### Desired behavior

It works without the `overwrite` option.
It also works with mochawesome v6.3.1. 

I couldn't reproduce with mocha and mochawesome alone (without cypress), so I suspect a Cypress issue. It does occur in all our Cypress projects which use mochawesome.

### Test code to reproduce

https://github.com/cexbrayat/mochawesome-v7

```
yarn
yarn test
```

Remove the overwrite option from `cypress.json` and it works again.

### Cypress Version

8.7.0

### Other

node v14.18.1
macOS 11.6

**Comment by brandonb927** [comment]
Came here as I also ran into this issue this morning when attempting to upgrade to mochawesome reporter v7. Exact same error and outcome.

**Comment by cexbrayat** [comment]
Was fixed in mochawesome v7.0.1
